# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/RangeAnalysis.cpp
# Commit: cb4534b7dbb7
# Full Hash: cb4534b7dbb746d083bdbf8fb74b3036c63f897b
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-10-02 04:10:47
# Description:
#   Bug 1668197: Handle OOM in MHypot::New r=jandem
#   
#   In addition to the calls in the transpiler, MHypot::New is also called in the MHypot `clone` implementation.
#   
#   I checked the other fallible `New` methods, and it looks like we are handling everything else correctly.
# ==============================================================================

diff -r 66666ee5b0a9 -r cb4534b7dbb7 js/src/jit/RangeAnalysis.cpp
--- a/js/src/jit/RangeAnalysis.cpp	Thu Oct 01 07:19:04 2020 +0000
+++ b/js/src/jit/RangeAnalysis.cpp	Thu Oct 01 16:32:15 2020 +0000
@@ -2861,6 +2861,9 @@
   }
 
   MInstruction* clone = candidate->clone(alloc, operands);
+  if (!clone) {
+    return false;
+  }
   clone->setRange(nullptr);
 
   // Set UseRemoved flag on the cloned instruction in order to chain recover